packages/roosterjs-content-model-api/lib/modelApi/common/queryContentModelBlocks.ts

```diff
@@ -0,0 +1,91 @@
+import type {
+    ContentModelBlockType,
+    ReadonlyContentModelBlock,
+    ReadonlyContentModelBlockGroup,
+} from 'roosterjs-content-model-types';
+
+/**
+ * Query content model blocks
+ * @param group The block group to query
+ * @param blockType The type of block to query @default 'Paragraph'
+ * @param filter Optional selector to filter the blocks
+ * @param findFirstOnly True to return the first block only, false to return all blocks
+ */
+export function queryContentModelBlocks<T extends ReadonlyContentModelBlock>(
+    group: ReadonlyContentModelBlockGroup,
+    blockType?: ContentModelBlockType,
+    filter?: (element: T) => element is T,
+    findFirstOnly?: boolean
+): T[] {
```

Comment:
Do you also want to return the path?

Reply:
No, all the elements of type or filtered by the filter function